jmclient/jmclient/taker.py

```diff
@@ -485,12 +485,11 @@ def receive_utxos(self, ioauth_data):
         #we have tried to avoid this based on over-estimating the needed amount
         #in SendPayment.create_tx(), but it is still a possibility if one maker
         #uses a *lot* of inputs.
-        if self.my_change_addr and my_change_value <= 0:
+        if my_change_value < 0:
```

Comment:
I don't think this change is necessary. Note in particular that the sweep case is covered here:

https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/be6e4fd58fbb07339a205104cd82e97925027fe4/jmclient/jmclient/taker.py#L501-L508

Reply:
It is possible for `my_change_value` to be 0 even if we expected to use a change address. In that case we should just not use a change address rather than failing. That's the behavior changed here.

Reply:
> It is possible for my_change_value to be 0 even if we expected to use a change address. In that case we should just not use a change address rather than failing.

Yes, on reading again, I agree. We should not have the `=` case on that line, it folds into the following check for whether we throw away the change output because less than dust. The zero case is perfectly acceptable there.

> Further, if we are doing a sweep, my_change_value should be 0.

Right, it should be, but as per the extra code I quoted above, there is apparently a rounding error case (I don't know if this case actually still happens, it was something from the very start of the project to do with floating point calcs (@chris-belcher may conceivably remember this, as he wrote it right at the start, but it's a heck of a long time ago!), but I don't see a need to check and remove it, if we do get +1/-1 sat errors it's fine). Hence I think we need to leave in the `if self.my_change_addr` part unless that aspect is carefully checked again. Otherwise you will wrongly pick up sweeps for rejection with -1 (in that case, as per the comment, you just end with 1 sat less in the miner fee, which is fine).

Reply:
Oh, right, I forgot about the rounding thing.